cmd/zdb/zdb.c

```diff
@@ -6335,6 +6335,33 @@ zdb_vdev_lookup(vdev_t *vdev, const char *path)
 	return (NULL);
 }
 
+static int
+name_from_objset_id(spa_t *spa, char *instr, char *outstr)
+{
+	uint64_t objset_id;
+	dsl_dataset_t *ds;
+
+	objset_id = strtoull(instr, NULL, 0);
+	if (objset_id == 0 && strcmp(instr, "0") != 0 &&
+	    strcmp(instr, "0x0") != 0) {
+		fprintf(stderr, "invalid objset number %s\n", instr);
+		return (EINVAL);
+	}
+	dsl_pool_config_enter(spa->spa_dsl_pool, FTAG);
+	int error = dsl_dataset_hold_obj(spa->spa_dsl_pool, objset_id,
+	    NULL, &ds);
+	if (error != 0) {
+		(void) fprintf(stderr, "failed to hold objset %s: %s\n",
```

Comment:
Using `%d` and `objset_id` here in the error would be preferable so its clear how `strtoull` parsed this.